### ISSUE TITLE
Latency measurements

### ISSUE THREAD

**Issue by jpcima** [magnetophon/molot-lite#13]
I find 1.5 sample at 2x, and 2 samples at higher rates.
Where do you have these measurements from?

**Comment by magnetophon** [comment]
I measured them by duplicating and phaseflipping a track.
Then I just adjusted the latency compensation in Ardour and listened for the deepest null.

How did you measure?
Could it be that the latency is dependent on the host samplerate?
At which samplerate did you measure?
I used 48k.

In any case, it would be great if the latency could be reported to the host.

**Comment by jpcima** [comment]
Btw in README you wrote "0x", this must be "1x".

The measure is very lowlevel, it was a program which sends a sine wave into the oversampler.
I'm plotting this data in gnuplot and setting the appropriate x-offset such that sine waves align.
The samplerate of reference was 44.1k.
[Latency.cpp.gz](https://github.com/magnetophon/molot-lite/files/7344927/Latency.cpp.gz)

>Could it be that the latency is dependent on the host samplerate?

Really this latency might just be frequency-dependent phase shift, then there's not much that can be done about it.

**Comment by magnetophon** [comment]
It's almost certainly frequency dependent phase shift; you can hear different frequency ranges being canceled out at different latency-compensation values.
Also from a theoretical POV it makes sense that an anti-aliassing filter, as used in the oversampler, would have a phase shift, right?

Subjectively, the values in the README give the greatest cancelation over the widest frequency range.
Even if it is not in a strict sense a latency, how do you feel about reporting it to the host, so it can be compensated?

I've fixed the typo, thanks for the headsup!

What frequency is the sine-wave you are using?
I'll repeat my test with a sine at that freq.

**Comment by jpcima** [comment]
> Also from a theoretical POV it makes sense that an anti-aliassing filter, as used in the oversampler, would have a phase shift, right?

Yes, this oversampler uses some methods which are a bit out the ordinary, I'm not fully on the clear about how it works.
It uses FIR filters and these have phase shift. It would be probably more present, the higher the frequency of the input.

> how do you feel about reporting it to the host, so it can be compensated?

I don't have a correct answer, likely I'd leave as it is.

> What frequency is the sine-wave you are using?

Sine is 440Hz

**Comment by magnetophon** [comment]
> I don't have a correct answer, likely I'd leave as it is.

Fair enough. 

OTOH:
If it sounds right, it is right!  ;)

To my ears, a 5 sample latency compensation makes a big improvement in the amount of cancelation in the above scenario.
